Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -218,24 +175,22 @@ internal ExchangeBuffer(MemoryBuffer<T, Index1> buffer, ExchangeBufferMode mode)
         /// <summary>
         /// Returns the extent of this buffer.
         /// </summary>
-        public Index1 Extent { get; }
+        public TIndex Extent { get; protected set; }
 
         /// <summary>
         /// Returns an array view to the CPU part of this buffer.
         /// </summary>
-        public ArrayView<T> CPUView { get; }
+        public ArrayView<T, TIndex> CPUView { get; protected set; }
 
         /// <summary>
         /// Returns a reference to the i-th element in CPU memory.
         /// </summary>
         /// <param name="index">The element index to access.</param>
         /// <returns>A reference to the i-th element in CPU memory.</returns>
-        public unsafe ref T this[Index1 index]
+        public T this[TIndex index]
```

Comment:
nit: shouldn't this accessor return a reference to the i-th element, or is this load-by-value change intended?

Reply:
Oh, should I keep that as a reference? I guess I didn't update the comment, but if it's preferred to return a reference I can do that.

Reply:
I only changed this because it looked cleaner to me and I didn't understand the point of returning a reference here.

Reply:
I guess we should return a reference to leave further load-store optimizations to the JIT compiler.

Reply:
But I also agree with this piece of code 🤓

Reply:
Alright, sounds good. I'll return a reference.